cmake/config-unix/pyconfig.h.in

```diff
@@ -1160,7 +1160,13 @@
 #cmakedefine _GNU_SOURCE 1
 
 /* This must be defined on some systems to enable large file support. */
-#cmakedefine _LARGEFILE_SOURCE @_LARGEFILE_SOURCE@
+#cmakedefine _PY_LARGEFILE_SOURCE @_LARGEFILE_SOURCE@
+#if _LARGEFILE_SOURCE != _PY_LARGEFILE_SOURCE
+# error "_LARGEFILE_SOURCE is inconsistently defined"
```

Comment:
Moving forward I would like most of the definition in pyconfig to due a consistency check. 

Look like I missed something here, I basically need to create a macro and test it. 

I will keep this for later and integrate your patch as is.